### DISCUSSION TITLE
How should we position Jan?

### DISCUSSION THREAD

**Discussion by dan-menlo** [janhq/jan#1262] [General]
I'd like to create an open discussion on how we should position Jan. 

## Why is this a concern?

- Imho, Jan's product positioning and messaging is not strong
    - Our team is mostly product & engineering, and are not natural marketing people
    - We are especially weak at marketing towards "normies", as we have been on the bleeding edge for so long
- More importantly, I think there is a bigger need to educate consumers on Local and open-source AI
    - ChatGPT has entered the zeitgeist, while Local AI is only known by a group of nerds
    - How do we convince people to run AI on their own laptops?
    
## Where we are now

- Jan is a very young product, our first stable release was on 20th December
    - We have mostly grown from word-of-mouth, and our growth has come from a few r/localllama comments
    - We don't even have social media accounts set up yet (e.g. Twitter, Reddit) 
    - We haven't done a real announcement post (e.g. ProductHunt, Hackernews, r/localllama)

![image](https://github.com/janhq/jan/assets/101145494/f7001edb-6952-42e1-89a9-08249c0349cf)

## Our current value propositions

I would break down Jan's current "word-of-mouth" value propositions, by Crossing the Chasm populations. 

**Innovators**
- e.g. AI researchers, engineers: r/localllama, AI Engineering Twitter, AI Discords
    - Jan is not designed for this group, and I find the dynamics in this group very "competitive" (science club-like)
    - I would rather focus on educating the remaining 4.6 billion computer owners on what Local AI can do
- "You mean I can fork Jan and offer it to my customers commercially?"
    - That said, Jan can be used by "innovators" group in their businesses (e.g. AI consultancies)
    - AI consultants can deploy Jan for their customers (as an alternative to ChatGPT) and charge customers
    - AI consultants can (in the future) develop custom assistants on top of Jan, and let clients run the Assistants on their own Jan Desktops
    - [Not sure yet] AI consultants can optionally license Jan to whitelabel and close source it
   
**Early Adopters: AI Enthusiasts**
- e.g. Local AI users: r/localllama, AI enthusiasts
    - This is the surest bet, but there are several different tools available
- Value Prop: "Jan is an open source alternative to LMStudio" 
    - This is the nearest adjacent product, as "polished" Desktop Apps that allow users to run Local LLMs
    - However, I don't like this as I think open/closed source are both valid decisions, and imho LMStudio and Jan will have very different long-term niches
    - I dislike equating open source with virtue = we all need to somehow pay the bills, and in our case OSS aligns with our commercialization model
    - I actually think we should work with LMStudio, Ooba and others to grow the Local AI pie
- Value Prop: Jan is simpler to use than other tools
    - "I would recommend normal users use Jan instead of Ollama, because it's simpler"
    - "Oh nice, Jan is easier to use than Oobabooga... I can actually recommend this for my office" 
    - "Oh cool, you're just a 1-click install, no need for anyone to understand Docker" 
- Jan works across Windows, Mac and Linux seamlessly
    -  "Jan is an open source Desktop AI for you to try the latest LLMs and customize how they work"

<img width="879" alt="image" src="https://github.com/janhq/jan/assets/101145494/755f449e-3ff0-47fc-979f-4433f159c9f5">

<img width="620" alt="image" src="https://github.com/janhq/jan/assets/101145494/fe0e465e-66dd-450a-8a1c-fd9fe6196c78">

**Early Adopters: Developers**
- Value Prop: Drop-in replacement for OpenAI's API and platform
    - Developers just need to change `https://api.openai.com` to `localhost:1337`
    - We aim to offer full API compatibility, see [https://jan.ai/api-reference/](https://jan.ai/api-reference/)
    - Not just `/chat/completions` but the full API 
- Value Prop: I can experiment, tinker with Local LLMs to find one that is suitable for my app
    - Find a LLM that works well with reasoning tasks, logical tasks etc

**Early Majority: Consumers**
- e.g. "Normal people" who currently use ChatGPT, and are just realizing local and open source AI exists
- Value Prop: Local, offline alternative to ChatGPT 
    - "Jan is a local, private alternative to ChatGPT that runs only on your computer"
    - "Wow, you mean this works without the internet?"
- Value Prop: No subscription fees
    - "Wow, you mean I don't need to pay ChatGPT $20 a month?"
    - "Wow, you mean this is free?"
- Value Prop: Privacy
    - "Wow, you mean my data doesn't leave my computer?"
    - Hmm... <talks dirty to AI>
- Value Prop: 10x your productivity
    - This is our longer-term roadmap, and not really where we are now

**Early Majority: Businesses**
- e.g. We have a version of Jan that is a full "ChatGPT-alternative" that businesses can deploy on-prem or on their own AWS
    - Value Prop: Provide an self-hosted alternative to ChatGPT (i.e. privacy, data protection etc)
- e.g. This version of Jan can also be used to host a custom LLM, that is customised to their needs
    - Value Prop: Fine-grained control of their AI (i.e. works deterministically, not a black box)
    - Value Prop: Not beholden to OpenAI (e.g. downtime, policy changes, rate limits)

## Current Features

> Public Roadmap: https://github.com/orgs/janhq/projects/5/views/16

- Works on Windows, Linux, Mac
- Desktop App (doesn't require Docker, CUDA)
- Very simple to run Local AI
- 1-click model downloads
- API server that runs at localhost:1337 (OpenAI-compatible API)
- System Monitor (e.g. CPU, RAM, Hardware usage)
- Open Source (i.e. code can be audited prior to deployment)

One important note: we are having a bunch of teething issues on Windows, but these are being rapidly fixed and I anticipate Windows will be fairly stable by end-Jan. 

Upcoming features: 
- Jan Server (multi-user, for businesses)
- Desktop Extensions (i.e. Grammarly-like)
- Mobile Extensions
- Assistants Framework
- Extensions (similar to VSCode)

## Current Misconceptions

- Jan runs other open source AIs, and isn't an AI by itself
    - Models: OpenHermes, Mistral etc
    - We often have to explain to people we don't provide the AI, rather we're a tool for them to run popular Open Source AIs
- "Oh, you guys are just putting a UI"
    - Jan is building a full ecosystem and platform to support the adoption of open source AI 
    - We're going to build the tooling required to make open source a viable alternative to OpenAI/ChatGPT
    - Simple things like "don't crash on Windows 10", "no need to understand command line"
- "Oh, you guys are just a UI on top of llama.cpp"
    - Jan is actually a fully modular framework, similar to VSCode 
    - We support llama.cpp as one inference engine, but will be adding more soon (e.g. TensorRT-LLM, Apple's MLX)
    - Our goal is to make it possible for our mom to run local AI - not just at a UI level, but at a "automagical experience" engineering level
    
## Resources

- Why Jan? https://jan.ai/guides/
- Why does Jan exist? https://jan.ai/about/#why-does-jan-exist (outdated)

**Comment by Techeters** [comment]
One of the common failures of technical founders with marketing is that they explain what a product does, and not specifically what it does _for_ users.  When looking at these groups, we have to see what is important and common for innovators and early adopters:

ChatGPT has a cost
Privacy is an issue
ChatGPT is limited in capability
LLMs are great, but there are too many of them

So while everything in this document is accurate towards what Jan is great at and why it is relevant, the messaging can be fine tuned to more closely resemble the fact that most users/businesses don't want to pay for LLM, are highly concerned with privacy (especially for businesses and US-based users), ChatGPT is really restricted, and most users don't know which LLM is best to use.

"Jan is free to use, taps into multiple LLMs so you don't have to choose, has a wider range of capabilities than a single LLM, and is private and secure to your local server or computer"

**Reply by Techeters** [reply]
Does Jan have capabilities for enterprise clients and if not, how soon could it be ready for enterprise sales?

**Reply by JiveyGuy** [reply]
One use I think that Jan would be amazing for us local use in classified environments where use of a cloud LLM is impossible. This would allow rapid adoption of generative tools in these antiquated areas where even googling something is hard. Having access to a tool that you could ask simple things you would otherwise need to walk outside to learn would be a huge enterprise usecase with lots of funding.

**Comment by ghost** [comment]
I think Jan should position itself as a hub for AI tools that can be accessible from anywhere, including computers, phones, and browsers. There is already a lot of competition between huge companies, as they often end up copying and adding features that most start-ups come up with. And people just stop using these apps and prefer these big companies. However, the biggest disadvantage of these big companies is that they can only offer their own services.

This is where Jan jumps in. If we want Jan to not be affected by these companies and their almost unlimited money, Jan should offer users the ability to use any tool they want. 

Jan should offer; ChatGPT, Google Bard, Microsoft Co-Pilot / Dall-E, Stable Diffusion, or any LLM / MLLM that users want to download and use offline in one place.

I hope you can see my point. If Jan (or any other app) wants to survive in this harsh area, it should constantly integrate these tools and get benefit from. That will make Jan Future-proof. Otherwise, one day, a big tech company could release an app or add a feature that makes Jan functionless. If the core feature of Jan is running it locally and offline, there is a high possibility of losing its functionality in the future. 
Instead of advertising it as "ChatGPT Alternative, %100 Offline" it would be much better to advertise it as "an all-in-one alternative for using AI." that also runs locally.

So, my take on this discussion; Jan should be the ultimate tool that effortlessly/easily usable to regular users, offers detailed functionality to power users, and allows enterprises to build their own solutions with it.

**Comment by 0xgokuz** [comment]
> Our current mission is to allow humans and businesses to own their AI, with the right to tinker, repair and innovate.

Looking at your mission. I believe your positioning should be building the best developer tools to empower developers build and integrate A.I. easily however they want

So what does developer wants?
* Control - data, privacy, etc
* Open Platform - integrate, etc
* Can use whatever models they want - and combine, train, etc

Developers, Developers, Developers

![IMG_5758](https://github.com/janhq/jan/assets/155065525/da212952-c6b9-444e-83ae-a7e53cc1a3b6)

**Reply by thinwoodsman** [reply]
I know this is discussion is mostly for Jan developers, but I wanted to chime in on this particular use case to provide some deeper context.

I am new to Jan, and came to it recently after playing with LM Studio, and before playing with Local AI. From my perspective, these three tools mainly do the same thing: provide an accessible front-end to LLMs, as well as a drop-in replacement for the OpenAI API server. Of the three, LocalAI probably stands out the most in terms of features, while Jan and LM Studio are pretty even. Right off the bad, this positions Jan as "the open source LM Studio".

None of these three applications suits my needs because they all have the same basic flaw: the model has to be in GGUF format. I don't want to get in to the pros and cons of this. I have been writing command-line utilities in Python for using LLMs for a couple of years now, and for using PyTorch models a few years beyond that. I have a terabyte or two of models that are in GGUF, GGML, safetensors, and whatever the binary format is for PyTorch. I have already written my own local API server to wrap these, and while it is admittedly a baling-wire-and-duct-tape job, it will instantiate any format of model pretty seamlessly, allowing me to compare their performance quite easily. Yet neither Jan, nor LM Studio, nor Local AI can do the same. I'm sure there's a good reason, and I know that I could take the time to make or download GGUF files for these models, but what I am saying is that all three apps are losing me right off the bat.

What I would like to see is one of these apps designed along a standard plugin/extension architecture, where every component can be replaced by the end user or a third party: the model framework could be cortex or llama.cpp or transformers, the model itself could be torch or tensorflow or onnx, the model class or loader could be a Python class written by the user in order to hook the model or add layers. I say "designed" this way because only the current behavior of Jan/LMStudio/LocalAI needs to be implemented: the users can then take that and run crazy with it, as users are wont to do. [EDIT: I found the Cortex architecture document and it looks like a Python runtime extension is on it, which is good. Still reading the code for the extensions to see how they work] 

All of which is a long-winded way of restating the above points about what developers want: Control, Open platform, and Model choice.


As to non-developer use, the selling points of Jan are geared towards what people are now calling "private cloud" and "edge computing", and I think these have been discussed suitably well. Government clients (not just military) want control of the servers and the applications that run on them, the health care industry is similar but less adamant about it, and pretty much everyone in finance thinks their special sauce is supersekret, so those would be good use cases to develop demos/examples for.

Finally, to not end on a negative note, great job on Jan, it works well and I have had fun experimenting with it.

**Comment by TomLucidor** [comment]
The layering:
- The Open LLM models for Privacy - LM Studio curation did this well
- Smooth UX for usability - need to be multi-modal and handle multiple architectures
- RAG - Has to beat tools like Khoj and GPT4All (which can handle both documents and smart notebooks like LogSeq)
- (optional) "Auto" operation agents - be able to search the web for info and cache webpages in multiple ways
- (optional) Interaction and conversation - be able to take multiple points of view and act like ChatDev or MetaGPT or ChatArena

**Comment by Nierusan** [comment]
À maybe stupid comment but I tried several apps to test gpt and else and all of them miss one thing. A search into prompt.

So I have to launch new prompt everytime and sometimes I know I ask for it 20 times already but without a way to search I won't be able to find my previous prompt.

**Comment by FilterJoe** [comment]
Have been using Jan for just a few days. Jan is close to being usable by "Normies" but not quite there. The part that works really well for normies is:

1) Easy to install Jan
2) Easy to install a model if it's in Jan's curated hub (however, it's vastly too complex to do a user installation because the generated default JSON is not correct).
3) Easy to start a thread

This could be made even easier if a default model (For May/June 2024, probably Lllama 8B Q4) was already installed to begin with, and that was the default model when you type into a prompt.

Then the marketing message for normies would be easy:

Run a local LLM on your notebook that's better than ChatGPT 3.5. Install Jan, open it, and start chatting!

**Comment by ianscrivener** [comment]
It is great that you are asking this very important question!! ... and the have been some good comments.

To survive and thrive Jan needs a clear product value proposition and a strong brand.  

A lot of technical startup founders go through some kind of a startup incubation course/program... which will always include a lot of time spent workshopping and fine tuning the core brand values and product propositions. Simon Sinek's book **"Start with why"** is a great read... receommended.

There is kind of a 'bottom up' versus 'top down' difference/tension here. Developers (I'm one) are usually kind of bottom up.. built it and they will come. That is great and very necessary.

But what is also needed IMO is a kind of "Product Architect" thinking... what is the competitor product landscape... and where does Jan fit into that? This top down strategic thinking will help define the product features that will help Jan to survive and thrive!!

**Reply by TomLucidor** [reply]
The whole landscape is excessively packed now, that the best position is "people are the product" (or strong learning community and tech support), everything from exclusive models (Anthropic) to agentic workflows (check SWE-Bench for example) are already goner. There are not enough social infrastructure however. Food for thought.

**Comment by ksaadDE** [comment]
Now I understand why you ditched docker and want to sell with Jan Hosted soon. 

It is unlikely to succeed, as there are already Skypilot and other such platforms allowing auto-hosting in the cloud and locally, latter less.

The question would be how do you earn money through it? Probably you do not. The market is satisfied. Open a "spent a coffee" tab and let companies and people spent money and ask for spending at several businesses. There is no business case yet. 

On the other hand, its unlikely that I pay for it. I might buy a coffee or sponsor a feature or objective that is crucial for my workloads but I will not want to pay monthly or per user for it. Then I could just grab ChatGPT and pay for that. 🤷‍♂️ 

Also, I wanted to test Jan AI but there is no docker-compose so I will delay it, once you are finished with it. Once I find something similar its likely that I ditch your "product". Especially if the competitor product runs with docker, as literally everything and everyone uses Docker, especially for testing. Pulling it, running it, playing with it. Some larger environments use Kubernetes and some use even docker in Prod.   

AppImage nothing else than a .deb or whatever sys installer pack, but docker are containers. They are a bit more "split off" from the system, thus a tiny bit more secure. If I need more security, I will use a Virtual Machine. However, nothing is 100% secure and I do not run my tests to be "safe".

I run things quickly to understand what the technology does, to play around with it, and eventually to deploy it later on, or just to throw it in the bin, as it did not satisfy my current needs.

**Reply by TomLucidor** [reply]
If their business model becomes "paid tech support" for a FOSS product it would be much better. Like seriously, what other alternatives are out there that isn't enshittified?

**Reply by ksaadDE** [reply]
@TomLucidor  or something else extending that paradigm / financing model, yes.

The idea to let the other stuff circle around a core product isn't wrong at all, but in a highly satisfied environment questionable. Yet alone they can not provide more than others could, due to already pre-existing resource constraints.
